MetaMask/metamask-mobile
# feat: setting to show fiat values on testnets

## **Description**

Adds a setting toggle  to show fiat $ conversions on testnets.  It is off by default.  It also has a bottom sheet friction informing users about the setting.

The `selectConversionRate` selector  is modified to not return a fiat conversion rate when the setting is off.  This allows the setting to affect in all parts of the UI that access the conversion rate.

## **Related issues**

Translations PR: https://github.com/MetaMask/metamask-mobile/pull/9361

https://consensyssoftware.atlassian.net/jira/software/projects/MMASSETS/boards/640/backlog?assignee=63f30ca01223974bc049f942&selectedIssue=MMASSETS-190

## **Manual testing steps**

1. Switch to an eth testnet like sepolia, linea goerli, etc.
2. Verify no fiat $ values show in the UI
3. Toggle settings > advanced > show conversion on test networks = on
4. Verify that fiat $ values appear

## **Screenshots/Recordings**



### **Before**

https://github.com/MetaMask/metamask-mobile/assets/3500406/1ce5b5f5-62c2-4a08-ad6b-8623c5818de7


### **After**

https://github.com/MetaMask/metamask-mobile/assets/3500406/1c95bd2f-3a92-44d0-a9c5-6a1bda1a15ca


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
